internal/rest/resources/daemon.go

```diff
@@ -0,0 +1,51 @@
+package resources
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/rest/access"
+	"github.com/canonical/microcluster/rest/types"
+	"github.com/canonical/microcluster/state"
+)
+
+var daemonCmd = rest.Endpoint{
+	Path: "daemon/servers",
```

Comment:
As we don't allow changing the entire daemon's config file my thinking was there will be other endpoints in the future which allow changing a specific subset of the local daemon's config.

Reply:
Isn't every state change effectively a change to the daemon's configuration? What is special about the `daemon` component here other than the internal implementation detail of changing something in the `daemon` package?

Reply:
Ah sorry I read it the other way around naming it `/daemon` not `/servers`.

As the path is `/cluster/internal/servers` I had the feeling it makes more sense to also put `daemon` so that it's clear we are modifying the local daemon instead of any internal cluster setting.

Reply:
With `/cluster/internal` I am missing the bit that tells me I am modifying the actual daemon, not something internally that affects the whole cluster.

Let's take `/cluster/internal/sql` as an example. It allows you to run a query against the cluster DB but it's not daemon specific.

Reply:
That's actually a good point but I would argue `/sql` should be moved to `/cluster/sql` instead. 

I guess what I'm confused about is why is a "server" considered a daemon component, but the database is not?

Reply:
Are you saying that because it's a local-only action then it's a "daemon" component? But since we execute a cluster-wide hook when we run this endpoint then its not truly local-only.

Reply:
> Are you saying that because it's a local-only action then it's a "daemon" component? 

This.

As the hooks are triggered afterwards by calling `/cluster/internal/hooks/{type}` (which is not daemon specific) I guess that is fine.

I agree moving `/sql` one level up.